shared/teams/new-team/index.js

```diff
@@ -6,6 +6,37 @@ import {globalColors, globalMargins, globalStyles} from '../../styles'
 
 import type {Props} from './'
 
+// This logic is copied from go/protocol/keybase1/extras.go.
+
+const validTeamnamePart = (s: string): boolean => {
+  if (s.length < 2 || s.length > 16) {
+    return false
+  }
+
+  return /^([a-zA-Z0-9][a-zA-Z0-9_]?)+$/.test(s)
+}
+
+const validTeamname = (s: string): boolean => {
```

Comment:
🐑 🐑 you could also use this to disable the submit button, since it's already here

Reply:
I'm loathe to do that without also displaying an error to the user, otherwise the user might not realize why the submit button is greyed out. Added that to the TODO for displaying an error.

Reply:
Ah, good point. Sounds good